### PR TITLE
Merge main into version 2 branch

### PR DESCRIPTION
**Pull request by qsdigor**
This is mergeable branch of the `main` and `integration/version-2` branches

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,13 +18,13 @@ Include the library by adding an `implementation` reference to `dependencies` bl
 For [Java](https://mvnrepository.com/artifact/io.ably/ably-java/latest):
 
 ```groovy
-implementation 'io.ably:ably-java:2.0'
+implementation 'io.ably:ably-java:1.2.18'
```
Comment (KacperKluka) [inline README.md:21]: I think it should stay as `2.0` 😉

File: README.md [hunk 2]
```diff
@@ -18,13 +18,13 @@ Include the library by adding an `implementation` reference to `dependencies` bl
...
 ```
 
 For [Android](https://mvnrepository.com/artifact/io.ably/ably-android/latest):
 
 ```groovy
-implementation 'io.ably:ably-android:2.0'
+implementation 'io.ably:ably-android:1.2.18'
```
Comment (KacperKluka) [inline README.md:27]: The same here, it should be `2.0` 😉

**Review by KacperKluka — CHANGES_REQUESTED** [review]
A few suggestions 😁 
Also, there are a few empty files such as `maven.gradle` or `common.gradle`, why have they been added? 🤔

File: README.md (hunk 1 above)
Comment (qsdigor) [reply README.md:21]: Yes, it will be 2.0.0 per semantics versioning but we can do that before merging v2 into the main.

Comment (qsdigor) [reply README.md:21]: Gradle files may have slipped in from merging main. I have removed it https://github.com/ably/ably-java/pull/849/commits/d9c876b4e64150e1150b3a4e78b62d3cc56ee27a

File: README.md (hunk 2 above)
Comment (qsdigor) [reply README.md]: Here you go https://github.com/ably/ably-java/pull/849/commits/ac1222ee49e9ebb6941100f708369723df8afc1a

**Review by KacperKluka — APPROVED** [review]
LGTM 👍

**Pull request merged**